packages/lib/src/interactions/AxialSelectToZoom.tsx

```diff
@@ -15,26 +14,16 @@ interface Props extends CommonInteractionProps {
 function AxialSelectToZoom(props: Props) {
   const { axis, modifierKey, disabled } = props;
 
-  const { canvasSize, visRatio } = useVisCanvasContext();
-  const moveCameraTo = useMoveCameraTo();
+  const { visRatio } = useVisCanvasContext();
+  const zoomOnBox = useZoomOnBox();
 
-  const { width, height } = canvasSize;
-  const camera = useThree((state) => state.camera);
-
-  function onSelectionEnd(selection: Selection) {
-    const [worldStart, worldEnd] = selection.world;
+  function onSelectionEnd({ world: worldSelection }: Selection) {
+    const [worldStart, worldEnd] = worldSelection;
     if (worldStart.x === worldEnd.x || worldStart.y === worldEnd.y) {
       return;
     }
 
-    const zoomRect = getEnclosedRectangle(worldStart, worldEnd);
-    const { center: zoomRectCenter } = zoomRect;
-
-    // Change scale first so that moveCameraTo computes the updated camera bounds
-    camera.scale.set(zoomRect.width / width, zoomRect.height / height, 1);
-    camera.updateMatrixWorld();
-
-    moveCameraTo(zoomRectCenter);
+    zoomOnBox(Box.fromPoints(...worldSelection));
```

Comment:
See `useZoomOnBox` for implementation. The scale update logic doesn't change; I just remove `camera.updateMatrixWorld();` and access `box.size` and `box.center` instead of calling `getEnclosedRectangle`.